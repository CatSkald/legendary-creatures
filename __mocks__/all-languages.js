import languages from "../src/i18n/languages";

exports.allLanguages = Object.entries(languages).map(
  ([lang, langProps]) => langProps,
);
