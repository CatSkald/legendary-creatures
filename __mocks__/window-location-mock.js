exports.mockWindowLocation = () => {
  const originalWindowLocation = window.location;

  beforeEach(() => {
    delete window.location;

    // mock window.location
    window.location = Object.defineProperties(
      {},
      {
        ...Object.getOwnPropertyDescriptors(originalWindowLocation),
        assign: {
          configurable: true,
          value: jest.fn(),
        },
      },
    );
  });

  afterEach(() => {
    // restore original `jsdom` location
    window.location = originalWindowLocation;
  });
};
