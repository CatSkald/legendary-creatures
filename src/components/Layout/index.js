import "../../styles/global.scss";
import "../../styles/theme-light.scss";
import "../../styles/theme-dark.scss";
import styles from "./index.module.scss";

import React, { useState } from "react";
import Header from "../Header";
import Footer from "../Footer";

const LocaleContext = React.createContext();

const Layout = ({ children, pageContext: { language } }) => {
  const [isDarkColorTheme, switchColorTheme] = useState(false);

  function handleToggleColorTheme() {
    switchColorTheme(!isDarkColorTheme);
  }

  return (
    <LocaleContext.Provider value={{ language }}>
      <div
        className={`body ${styles.Wrapper} ${
          isDarkColorTheme ? "theme-dark" : "theme-light"
        }`}
      >
        <Header
          isDarkColorTheme={isDarkColorTheme}
          handleToggleColorTheme={handleToggleColorTheme}
        />
        <section className={styles.SiteContent} role="main">
          <div className={styles.Container}>{children}</div>
        </section>
        <Footer />
      </div>
    </LocaleContext.Provider>
  );
};

export { Layout, LocaleContext };
