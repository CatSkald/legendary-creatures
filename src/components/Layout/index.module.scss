.layout__container {
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}

.layout {
  display: flex;
  flex: 1;
  width: 100%;
}

.layout__content {
  width: 100%;
  max-width: var(--container-width);
  margin: 0 auto;
  padding: var(--space-sm);

  @media screen and (min-width: 768px /*--screen-medium*/) {
    padding: var(--space);
  }
}
