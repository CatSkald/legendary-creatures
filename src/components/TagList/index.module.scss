.TagList {
  a {
    border-radius: 3px;
    text-align: center;
    text-decoration: none;
    padding: var(--space-sm);
    display: inline-block;
  }
}

.TagListItem {
  margin-bottom: var(--space-sm);
}

.Tag {
  margin: var(--space) var(--space-sm);
  padding: var(--space-sm);
  font-weight: bold;

  background-color: var(--card-bg); /* Green */
  color: var(--card-link);

  &:hover {
    color: var(--card-link-hover); /* Green */
  }
}

.TagValue {
  margin: var(--space-sm);
  margin-left: var(--space);
  border: solid;
  border-width: 0.1rem;
  border-color: var(--card-bg);
  color: var(--content-link);

  &:hover {
    color: var(--content-link-hover); /* Green */
    border-color: var(--content-link-hover);
  }
}
