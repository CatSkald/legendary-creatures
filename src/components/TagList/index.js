import styles from "./index.module.scss";

import React from "react";
import { navigate, Link } from "gatsby";
import PropTypes from "prop-types";
import { LocaleContext } from "../Layout";

const TagList = props => {
  const { language } = React.useContext(LocaleContext);
  const { getTagUrl, getTagValueUrl } = require("../../utils/url-helpers");

  return (
    <section className={styles.TagList}>
      {Object.entries(props.tags).map(([tag, values], tagIndex) => {
        var tagUrl = getTagUrl(language.path, tag);
        return (
          <>
            <Link
              className={styles.Tag}
              key={tagIndex + tag}
              to={tagUrl}
              hrefLang={language.code}
              onClick={() => navigate(tagUrl)}
            >
              {tag}
            </Link>
            {values.map((value, index) => {
              const url = getTagValueUrl(language.path, tag, value);
              return (
                <Link
                  className={styles.TagValue}
                  key={`${tag}${tagIndex}_${index}${value}`}
                  to={url}
                  hrefLang={language.code}
                  onClick={() => navigate(url)}
                >
                  {value}
                </Link>
              );
            })}
            <br />
          </>
        );
      })}
    </section>
  );
};

TagList.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export const query = graphql`
  query TagValues($locale: String!, $tag: MarkdownRemarkFieldsEnum!) {
    allMarkdownRemark(
      filter: {
        fields: { locale: { eq: $locale } }
        frontmatter: { page: { eq: null } }
      }
    ) {
      distinct(field: $tag)
    }
  }
`;

export default TagList;
