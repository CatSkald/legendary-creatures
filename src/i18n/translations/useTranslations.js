import React from "react";
import { useStaticQuery, graphql } from "gatsby";
import { LocaleContext } from "../../components/Layout";

function useTranslations() {
  const { language } = React.useContext(LocaleContext);
  const { rawData } = useStaticQuery(queryTranslations);

  const translationsForCurrentLocale = rawData.edges
    .filter(item => item.node.name === language.code)
    .map(item => item.node.translations)[0];

  return translationsForCurrentLocale;
}

export default useTranslations;

const queryTranslations = graphql`
  query queryTranslations {
    rawData: allFile(filter: { sourceInstanceName: { eq: "translations" } }) {
      edges {
        node {
          name
          translations: childTranslationsYaml {
            Description
            Title
            ShortTitle
            NotFound404Header
            NotFound404Content
            footerMoto
            by
            for
            fromPerson
            fromPlace
            of
            Categories
            View
            all
            orSearchBy
            tags
            All
            Home
            Latest
            Next
            Prev
            Tags
            plural
            Number
            Origin
            Habitat
            CreaturesPageTitle
            TagsPageTitle
            origin
            categories
            number
            habitat
          }
        }
      }
    }
  }
`;
