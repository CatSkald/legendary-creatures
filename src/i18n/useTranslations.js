import React from "react";
import { useStaticQuery, graphql } from "gatsby";
import { LocaleContext } from "../components/Layout";

function useTranslations() {
  const { locale } = React.useContext(LocaleContext);
  const { rawData } = useStaticQuery(queryTranslations);

  const translationsForCurrentLocale = rawData.edges
    .filter(item => item.node.name === locale)
    .map(item => item.node.translations)[0];

  return translationsForCurrentLocale;
}

export default useTranslations;

const queryTranslations = graphql`
  query queryTranslations {
    rawData: allFile(
      filter: { sourceInstanceName: { eq: "translations" } }
    ) {
      edges {
        node {
          name
          translations: childTranslationsYaml {
            Description
            Title
            NotFound404Header
            NotFound404Content
            footerMoto
            Icon
            by
            for
            fromPerson
            fromPlace
            of
            using
            Categories
            All
            Home
            Latest
            Next
            Prev
            Tags
            plural
            Number
            Origin
            Habitat
          }
        }
      }
    }
  }
`;
