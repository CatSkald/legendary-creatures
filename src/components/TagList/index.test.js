import React from "react";
import { render } from "@testing-library/react";
import { LocaleContext } from "../Layout";

import TagList from "../TagList";
import useTranslations from "../../i18n/translations/useTranslations";

import { mockWindowLocation } from "../../../__mocks__/window-location-mock";

mockWindowLocation();
jest.mock("../../i18n/translations/useTranslations");

useTranslations.mockImplementation(() => ({
  tag1: "tag 1",
  tag2: "tag 2",
  tag3: "tag 3",
}));

describe("TagList", () => {
  test("renders correctly", () => {
    const tags = {
      tag1: ["value1", "value2"],
      tag2: ["value3"],
      tag3: [],
    };
    const language = { code: "en" };
    const { container } = render(
      <LocaleContext.Provider value={{ language }}>
        <TagList tags={tags} />,
      </LocaleContext.Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
