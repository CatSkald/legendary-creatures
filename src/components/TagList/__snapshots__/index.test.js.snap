// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList renders correctly for language {
  name: 'Українська',
  code: 'uk',
  path: '/uk',
  dateFormat: 'DD.MM.YYYY'
} 1`] = `
<div>
  <section
    class="tagList"
  >
    <div
      class="tagList__item"
    >
      <a
        class="tagList__tag"
        href="/uk/мітки#tag1"
        hreflang="uk"
      >
        tag 1
      </a>
      <br />
      <a
        class="tagList__value"
        href="/uk/search/tag1/value1"
        hreflang="uk"
      >
        value1
      </a>
      <a
        class="tagList__value"
        href="/uk/search/tag1/value2"
        hreflang="uk"
      >
        value2
      </a>
      <br />
    </div>
    <div
      class="tagList__item"
    >
      <a
        class="tagList__tag"
        href="/uk/мітки#tag2"
        hreflang="uk"
      >
        tag 2
      </a>
      <br />
      <a
        class="tagList__value"
        href="/uk/search/tag2/value3"
        hreflang="uk"
      >
        value3
      </a>
      <br />
    </div>
    <div
      class="tagList__item"
    >
      <a
        class="tagList__tag"
        href="/uk/мітки#tag3"
        hreflang="uk"
      >
        tag 3
      </a>
      <br />
      <br />
    </div>
  </section>
</div>
`;

exports[`TagList renders correctly for language { name: 'English', code: 'en', path: '', dateFormat: 'DD/MM/YYYY' } 1`] = `
<div>
  <section
    class="tagList"
  >
    <div
      class="tagList__item"
    >
      <a
        class="tagList__tag"
        href="/tags#tag1"
        hreflang="en"
      >
        tag 1
      </a>
      <br />
      <a
        class="tagList__value"
        href="/search/tag1/value1"
        hreflang="en"
      >
        value1
      </a>
      <a
        class="tagList__value"
        href="/search/tag1/value2"
        hreflang="en"
      >
        value2
      </a>
      <br />
    </div>
    <div
      class="tagList__item"
    >
      <a
        class="tagList__tag"
        href="/tags#tag2"
        hreflang="en"
      >
        tag 2
      </a>
      <br />
      <a
        class="tagList__value"
        href="/search/tag2/value3"
        hreflang="en"
      >
        value3
      </a>
      <br />
    </div>
    <div
      class="tagList__item"
    >
      <a
        class="tagList__tag"
        href="/tags#tag3"
        hreflang="en"
      >
        tag 3
      </a>
      <br />
      <br />
    </div>
  </section>
</div>
`;

exports[`TagList renders correctly for language { name: 'Русский', code: 'ru', path: '/ru', dateFormat: 'DD.MM.YYYY' } 1`] = `
<div>
  <section
    class="tagList"
  >
    <div
      class="tagList__item"
    >
      <a
        class="tagList__tag"
        href="/ru/метки#tag1"
        hreflang="ru"
      >
        tag 1
      </a>
      <br />
      <a
        class="tagList__value"
        href="/ru/search/tag1/value1"
        hreflang="ru"
      >
        value1
      </a>
      <a
        class="tagList__value"
        href="/ru/search/tag1/value2"
        hreflang="ru"
      >
        value2
      </a>
      <br />
    </div>
    <div
      class="tagList__item"
    >
      <a
        class="tagList__tag"
        href="/ru/метки#tag2"
        hreflang="ru"
      >
        tag 2
      </a>
      <br />
      <a
        class="tagList__value"
        href="/ru/search/tag2/value3"
        hreflang="ru"
      >
        value3
      </a>
      <br />
    </div>
    <div
      class="tagList__item"
    >
      <a
        class="tagList__tag"
        href="/ru/метки#tag3"
        hreflang="ru"
      >
        tag 3
      </a>
      <br />
      <br />
    </div>
  </section>
</div>
`;
